### PR TITLE
create migrations for file embeddings 

### LINKED ISSUE #54679: Create database table for keeping track of uploads

### PR DESCRIPTION
**Pull request by BolajiOlajide**
Closes #54679

This PR creates the database tables needed to track file uploads for the worker to pick up and act on. 
When a file has been successfully uploaded to the `blob store`, we'll add a record for it to be picked up by the embedding worker in the `file_embedding_jobs` table.

## Test plan

* Run `sg migrate up`
 The database tables `file_embedding_jobs` and `file_embedding_jobs_stats` should be created in your `sourcegraph` database.

* Run `sg migrate undo -db=frontend`
This should revert the creation of the tables above.

### CODE REVIEW

File: migrations/frontend/1688654650_create_file_embeddings_jobs_stats_table/up.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE IF NOT EXISTS file_embedding_job_stats (
```
Comment (sashaostrikov) [inline migrations/frontend/1688654650_create_file_embeddings_jobs_stats_table/up.sql:1]: TBH, I'd merge 2 migrations into 1, WDYT?

**Review by sashaostrikov — COMMENTED** [review]
I think we can merge the migrations 🤔

File: migrations/frontend/1688654650_create_file_embeddings_jobs_stats_table/up.sql (hunk 1 above)
Comment (BolajiOlajide) [reply migrations/frontend/1688654650_create_file_embeddings_jobs_stats_table/up.sql]: Done.

File: migrations/frontend/1688654272_create_file_embeddings_jobs_table/down.sql [hunk 2]
```diff
@@ -1 +1,2 @@
 DROP TABLE IF EXISTS file_embedding_jobs;
+DROP TABLE IF EXISTS file_embedding_job_stats;
```
Comment (sashaostrikov) [inline migrations/frontend/1688654272_create_file_embeddings_jobs_table/down.sql:2]: Should we delete this table first as it depends on the "jobs" table?

**Review by sashaostrikov — APPROVED** [review]
I think you need to run `sg generate` after latest changes, right?

File: migrations/frontend/1688654272_create_file_embeddings_jobs_table/up.sql [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+   process_after TIMESTAMP WITH TIME ZONE,
+   num_resets INTEGER NOT NULL DEFAULT 0,
+   num_failures INTEGER NOT NULL DEFAULT 0,
+   last_heartbeat_at TIMESTAMP WITH TIME ZONE,
+   execution_logs JSON [],
+   worker_hostname TEXT NOT NULL DEFAULT '',
+   cancel boolean DEFAULT FALSE NOT NULL,
+
+   -- file metadata columns
+   file_id TEXT NOT NULL, -- this should be text, depending on the blob store we use it might not always be numeric
```
Comment (pjlast) [inline migrations/frontend/1688654272_create_file_embeddings_jobs_table/up.sql:19]: Should this be a `file_id` or an `archive_id` ? We won't be embedding one file at a time, but rather all the files in the archive.

File: migrations/frontend/1688654272_create_file_embeddings_jobs_table/down.sql (hunk 2 above)
Comment (BolajiOlajide) [reply migrations/frontend/1688654272_create_file_embeddings_jobs_table/down.sql:1]: Done. I missed that.

Also,  I already ran `sg generate` earlier.

File: migrations/frontend/1688654272_create_file_embeddings_jobs_table/up.sql (hunk 3 above)
Comment (BolajiOlajide) [reply migrations/frontend/1688654272_create_file_embeddings_jobs_table/up.sql]: Updated!

**Pull request merged**